internal/controller/storageclient_controller.go

```diff
@@ -638,36 +802,49 @@ func (r *StorageClientReconciler) reconcileClientStatusReporterJob() (reconcile.
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClientReconciler) list(obj client.ObjectList, listOptions ...client.ListOption) error {
-	return r.Client.List(r.ctx, obj, listOptions...)
+func (r *StorageClientReconciler) hasPersistentVolumes() (bool, error) {
+	pvList := &corev1.PersistentVolumeList{}
+	if err := r.list(pvList, client.MatchingFields{pvClusterIDIndexName: r.storageClientHash}, client.Limit(1)); err != nil {
```

Comment:
Will this also works for the existing cephfs/rbd PV's (upgraded)? same is applicabel for other resouces like (VGSC,VGC etc)

Reply:
good catch, this is a bug in two ways
1. w/ convergence the clusterID will not be a hash but an UUID which is a bit difficult to get now, needs some thinking
2. your original assessment, ie, this skips pre-convergence PVs/VSCs/VGSCs

Reply:
Why can't we go by provisioner? is the concern that multiple clients will be using the same provisioner name. I am not sure that will be the case

Reply:
rechecked, there is no reference to provisioner in PV

Reply:
updated.

Reply:
>Why can't we go by provisioner? is the concern that multiple clients will be using the same provisioner name. I am not sure that will be the case

AFAIK it will be the same provisioner, Think about the case of RDR/Metro DR the provisioner names need to be the same for the PV's and also in ODF we never had concept of multiple provisioners (even for internal+external mode). If this is something we are planning to change we need a bigger discussion

Reply:
This is addressed now, not using the provisioner for identification